src/mongo_types.ts

```diff
@@ -516,23 +520,23 @@ export type NestedPaths<Type> = Type extends
   ? [string]
   : Type extends object
   ? {
-      [Key in Extract<keyof Type, string>]: Type[Key] extends Type // type of value extends the parent
-        ? [Key]
-        : // for a recursive union type, the child will never extend the parent type.
-        // but the parent will still extend the child
-        Type extends Type[Key]
-        ? [Key]
```

Comment:
This prevented template string keys from working because `` { a: Record<`example-${number}`, any> } extends Record<`example-${number}`, any> ``, which means this would output `['a']` instead of `['a', `example-${number}`]`, despite there being no recursion there.

This fact is exclusive to records with template string keys because template strings allow for infinite possible key values, which apparently informs TypeScript that it should allow any keys on the object.

That's why I had to change some of the recursive typing support here.